test/e2e/mcd_test.go

```diff
@@ -965,9 +965,16 @@ func TestMCDRotatesCertsOnPausedPool(t *testing.T) {
 	err = helpers.WaitForMCDToSyncCert(t, cs, selectedNode, controllerConfig.ResourceVersion)
 	require.Nil(t, err)
 	require.NotEmpty(t, nodes)
-	onDiskCert := helpers.ExecCmdOnNode(t, cs, selectedNode, "cat", "/rootfs/etc/kubernetes/kubelet-ca.crt")
 
-	assert.Equal(t, inClusterCert, onDiskCert)
+	// Due to the nature of the cert updates(more info in https://github.com/openshift/machine-config-operator/pull/3718)
+	// attempt to retry on failure after waiting for a few seconds before failing the test
+
+	var onDiskCert string
+	wait.PollImmediate(5*time.Second, 15*time.Second, func() (bool, error) {
+		onDiskCert = helpers.ExecCmdOnNode(t, cs, selectedNode, "cat", "/rootfs/etc/kubernetes/kubelet-ca.crt")
+		return (onDiskCert == inClusterCert), nil
```

Comment:
I'm good with this fix as it stands, but was wondering about your thoughts on re-fetching the inClusterCert as well in this test?

Just on the off chance by some accident that that changes so the onDiskCert got past the original version. In theory, they should eventually steady-state with each other

Reply:
That makes sense to me, to be extra safe! Also updated the comment and move a couple of things around so its cleaner (: